homeassistant/components/lyric/sensor.py

```diff
@@ -0,0 +1,254 @@
+                )
+            if device.displayedOutdoorHumidity:
+                entities.append(
+                    LyricOutdoorHumiditySensor(hass, coordinator, location, device)
+                )
+            if device.changeableValues:
+                if device.changeableValues.nextPeriodTime:
+                    entities.append(
+                        LyricNextPeriodSensor(hass, coordinator, location, device)
+                    )
+                if device.changeableValues.thermostatSetpointStatus:
+                    entities.append(
+                        LyricSetpointStatusSensor(hass, coordinator, location, device)
+                    )
+
+    async_add_entities(entities, True)
+
+
+class LyricSensor(LyricDeviceEntity):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+        key: str,
+        name: str,
+        icon: str,
+        device_class: str = None,
+        unit_of_measurement: str = None,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+        self._device_class = device_class
+        self._unit_of_measurement = unit_of_measurement
+
+        super().__init__(coordinator, location, device, key, name, icon)
+
+    @property
+    def device_class(self) -> str:
+        """Return the device class of the sensor."""
+        return self._device_class
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit this state is expressed in."""
+        return self._unit_of_measurement
+
+
+class LyricIndoorTemperatureSensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_indoor_temperature",
+            "Indoor Temperature",
+            "mdi:thermometer",
+            DEVICE_CLASS_TEMPERATURE,
+            hass.config.units.temperature_unit,
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self.device.indoorTemperature
+
+
+class LyricOutdoorTemperatureSensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_outdoor_temperature",
+            "Outdoor Temperature",
+            "mdi:thermometer",
```

Comment:
```suggestion
            None,
```

Let the UI pick the icon if you set the device class